crates/rattler_solve/src/resolvo/mod.rs

```diff
@@ -543,7 +577,53 @@ impl<'a> DependencyProvider for CondaDependencyProvider<'a> {
             dependencies.requirements.push(version_set_id.into());
```

Comment:
I think you need to split each matchspec into multiple solver specs if you have optional dependencies. E.g. the spec (in python terms) `boltons[foo, bar]` should result in 3 requirements: boltons, boltons with `foo`, and boltons with `bar`. Because one requirement can only ever match a single solvable, so to get the 3 solvables mentioned above you also need to add 3 requirements.